sdk/sqlmanagement/Azure.ResourceManager.Sql/src/Customized/Models/SqlMetricDefinition.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable disable
+
+using System;
+using System.ComponentModel;
+
+namespace Azure.ResourceManager.Sql.Models
+{
+    public partial class SqlMetricDefinition
+    {
+        /// <summary> Uri of the resource. </summary>
+        [EditorBrowsableAttribute(EditorBrowsableState.Never)]
+        [ObsoleteAttribute("This property has been replaced by ResourceUriString", false)]
+        public Uri ResourceUri { get; set; }
```

Comment:
Originally this is a read-only property. Shall we remove the setter here?
Besides, this property will be null after the model is deserialized. Probably need to read from `ResourceUriString`?
```suggestion
        public Uri ResourceUri => string.IsNullOrEmpty(ResourceUriString) ? null : new Uri(ResourceUriString);
```

Reply:
fixed